src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -7740,9 +7740,10 @@ void endRequestInternal() throws SQLException {
                 sqlWarnings = originalSqlWarnings;
                 if (null != openStatements) {
                     while (!openStatements.isEmpty()) {
-                        try (Statement st = openStatements.get(0)) {}
+                        Statement st = openStatements.get(0);
+                        st.close();
+                        removeOpenStatement((SQLServerStatement) st);
```

Comment:
I think it would be safer to keep the try with resources as that would guarantee close even if thread dies. Or we need to add a finally

Reply:
that's possible, removeOpenStatement can be in the finally

Reply:
done